Spaces in `PKG_SHA1` makevar cause Julia to attempt `rm -r /`

Thank God for systems that won't let you blithely run `rm -r /`.  Easy reproducer:

* Edit stdlib/Pkg.version to have a space at the end of `PKG_SHA1`
* Run `make -C stdlib Pkg`

You can then observe our build system attempt to `rm -r /`, then unpack the Pkg tarball into `/`:

```
$ make -C stdlib install-Pkg
make: Entering directory '/home/sabae/src/julia/stdlib'
Makefile:25: warning: overriding recipe for target 'Pkg-a4aaae26d7724d7ec1440aac5c063865ecba9503'
Makefile:25: warning: ignoring old recipe for target 'Pkg-a4aaae26d7724d7ec1440aac5c063865ecba9503'
make: Circular Pkg-a4aaae26d7724d7ec1440aac5c063865ecba9503 <- Pkg-a4aaae26d7724d7ec1440aac5c063865ecba9503 dependency dropped.
/home/sabae/src/julia/deps/tools/jldownload /home/sabae/src/julia/stdlib/srccache/Pkg-a4aaae26d7724d7ec1440aac5c063865ecba9503 https://api.github.com/repos/JuliaLang/Pkg.jl/tar$
all/a4aaae26d7724d7ec1440aac5c063865ecba9503
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   395  100   395    0     0   1238      0 --:--:-- --:--:-- --:--:--  1234
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 2032k    0 2032k    0     0  2155k      0 --:--:-- --:--:-- --:--:-- 2155k
/home/sabae/src/julia/deps/tools/jldownload .tar.gz https://api.github.com/repos/JuliaLang/Pkg.jl/tarball/a4aaae26d7724d7ec1440aac5c063865ecba9503
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   497  100   497    0     0    537      0 --:--:-- --:--:-- --:--:--   536
100 2032k  100 2032k    0     0  1177k      0  0:00:01  0:00:01 --:--:-- 3450k
/home/sabae/src/julia/deps/tools/jlchecksum /home/sabae/src/julia/stdlib/srccache/Pkg-a4aaae26d7724d7ec1440aac5c063865ecba9503
WARNING: sha512 checksum for Pkg-a4aaae26d7724d7ec1440aac5c063865ecba9503 not found in deps/checksums/, autogenerating...
[ ! \( -e / -o -h / \) ] || rm -r /
rm: it is dangerous to operate recursively on '/'
rm: use --no-preserve-root to override this failsafe
make: [Makefile:26: /source-extracted] Error 1 (ignored)
mkdir -p /
/usr/bin/tar -C / --strip-components 1 -xf /home/sabae/src/julia/stdlib/srccache/Pkg-a4aaae26d7724d7ec1440aac5c063865ecba9503
```

--- a/deps/tools/git-external.mk
+++ b/deps/tools/git-external.mk
@@ -22,6 +22,8 @@
 #
 define git-external
 include $$(SRCDIR)/$1.version
+$2_SHA1 := $$(strip $$($2_SHA1))
+$2_BRANCH := $$(strip $$($2_BRANCH))
 
 ifneq (,$$(filter $1 1,$$(DEPS_GIT)))
 $2_SRC_DIR := $1

